sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSonarWayProfile.java

```diff
@@ -23,19 +23,37 @@
 import java.lang.reflect.Method;
 import java.util.HashSet;
 import java.util.Set;
+import org.sonar.api.SonarRuntime;
 import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.api.utils.Version;
 import org.sonar.api.utils.log.Logger;
 import org.sonar.api.utils.log.Loggers;
 import org.sonarsource.analyzer.commons.BuiltInQualityProfileJsonLoader;
 
 public class CSharpSonarWayProfile implements BuiltInQualityProfilesDefinition {
   private static final Logger LOG = Loggers.get(CSharpSonarWayProfile.class);
+  private static final Version SQ_7_4 = Version.create(7, 4);
+  private final boolean supportsSecurityFrontend;
+
+  public CSharpSonarWayProfile(SonarRuntime sonarRuntime) {
+    this.supportsSecurityFrontend = sonarRuntime.getApiVersion().isGreaterThanOrEqual(SQ_7_4);
+  }
 
   @Override
   public void define(Context context) {
     NewBuiltInQualityProfile sonarWay = context.createBuiltInQualityProfile("Sonar way", CSharpPlugin.LANGUAGE_KEY);
     BuiltInQualityProfileJsonLoader.load(sonarWay, CSharpPlugin.REPOSITORY_KEY, "org/sonar/plugins/csharp/Sonar_way_profile.json");
-    getSecurityRuleKeys().forEach(key -> sonarWay.activateRule(CSharpPlugin.REPOSITORY_KEY, key));
+    final String repositoryKey;
+    if (supportsSecurityFrontend) {
+      repositoryKey = "roslyn.sonaranalyzer.security.cs";
```

Comment:
It would be better if we retrieve the repository key from the SonarSecurity's class through reflection. In case when SonarSecurity is old version and the method returning the repository key is missing we should fallback to SonarC#'s repository key (csharpsquid)